src/org/zaproxy/zap/extension/formhandler/FormHandlerParam.java

```diff
@@ -0,0 +1,233 @@
+
+        ((HierarchicalConfiguration) getConfig()).clearTree(ALL_TOKENS_KEY);
+
+        ArrayList<String> enabledFields = new ArrayList<>(fields.size());
+        for (int i = 0, size = fields.size(); i < size; ++i) {
+            String elementBaseKey = ALL_TOKENS_KEY + "(" + i + ").";
+            FormHandlerParamField field = fields.get(i);
+
+            getConfig().setProperty(elementBaseKey + TOKEN_NAME_KEY, field.getName().toLowerCase());
+            getConfig().setProperty(elementBaseKey + TOKEN_VALUE_KEY, field.getValue());
+            getConfig().setProperty(elementBaseKey + TOKEN_ENABLED_KEY, Boolean.valueOf(field.isEnabled()));
+
+            if (field.isEnabled()) {
+                enabledFields.add(field.getName().toLowerCase());
+            }
+        }
+
+        enabledFields.trimToSize();
+        this.enabledFieldsNames = enabledFields;
+    }
+
+    /**
+     * Adds a new field with the given {@code name} and {@code value}, enabled by default.
+     * <p>
+     * The call to this method has no effect if the given {@code name} is null or empty, or a field with the given name already
+     * exist.
+     *
+     * @param name the name of the field that will be added
+     * @param value the value of the field that will be added
+     */
+    public void addField(String name, String value) {
+        if (name == null || name.isEmpty()) {
+            return;
+        }
+
+        for (Iterator<FormHandlerParamField> it = fields.iterator(); it.hasNext();) {
+            if (name.equalsIgnoreCase(it.next().getName())) {
+                return;
+            }
+        }
+
+        this.fields.add(new FormHandlerParamField(name, value));
+
+        this.enabledFieldsNames.add(name);
+    }
+
+    /**
+     * Removes the field with the given {@code name}.
+     * <p>
+     * The call to this method has no effect if the given {@code name} is null or empty, or a field with the given {@code name}
+     * does not exist.
+     *
+     * @param name the name of the field that will be removed
+     */
+    public void removeField(String name) {
+        if (name == null || name.isEmpty()) {
+            return;
+        }
+
+        for (Iterator<FormHandlerParamField> it = fields.iterator(); it.hasNext();) {
+            FormHandlerParamField field = it.next();
+            if (name.equals(field.getName())) {
+                it.remove();
+                if (field.isEnabled()) {
+                    this.enabledFieldsNames.remove(name);
+                }
+                break;
+            }
+        }
+    }
+
+    /**
+     * Gets the value for the field {@code name} by searching through all existing fields
+     *
+     * @param name the name of the field being queried.
+     * @return the value of the field
+     */
+    public String getField(String name){
+        this.setFields(fields);
+        for (FormHandlerParamField field: fields){
+            if(field.getName().equals(name)){
```

Comment:
It should ensure the equality is case insensitive (while `getName()` is all lower case `name` might not be), we should either use `equalsIgnoreCase` or lower case the `name` before the check.